### DISCUSSION TITLE
Images from base 64 encoded data

### DISCUSSION THREAD

**Discussion by alexiskhoury** [ocornut/imgui#5953] [New Users Build/Link/Run issues ONLY!]
Hi,
I am building an IoT dashboard using ImGui as graphical interface (thanks for its simplicity) but wondering if the library have or will implement the following:

1- More graphical widgets (less important): e.g. gauges; donut/pie charts, etc.
2- Rendering imges from base64 binary data: in fact my application uses MQTT to retreive data, and I can publish images using base64 encoding to the Mqtt server. The idea is the get this binary data and render the image using ImGui. I have seen that rendering images from local files is quite complex with ImGui (my application has already the Mqtt client code to subscribe to the desired topics)

Any help, or ideas are appreciated. below a screenshot of the development achieved so far...

<img width="647" alt="image" src="https://user-images.githubusercontent.com/17233706/205489608-31e223f1-2e04-43c7-aa1a-fe8c3baf328e.png">

Thanks in advance!

**Comment by ocornut** [comment]
1. check wiki for list of known extension. 
2. question has nothing to do with dear imgui and won’t be discussed here. Read wiki about textures loading and displaying example.

**Comment by PathogenDavid** [comment]
The process of uploading images to the GPU is dependent on the rendering backend you're using. There are [examples for several backends on the wiki](https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples).

All the others parts are honestly outside the scope of this forum, but the general process would be something like:

1. Download your image from MQTT
2. Decode the Base64
3. Decode the actual image (since it's presumably a PNG or a JPEG or something - [`stb_image`](https://github.com/nothings/stb) is a simple library for doing this)
4. [Upload the image to the GPU](https://github.com/ocornut/imgui/wiki/Image-Loading-and-Displaying-Examples)
5. Show the image using the platform-specific `ImTextureID` using `ImGui::Image`, `ImGui::ImageButton`, or by submitting it to an `ImDrawList` directly.

----------------

As an aside, the discussions section is for problems with building/running Dear ImGui and fonts only. Please [open an issue](https://github.com/ocornut/imgui/issues/new) next time.

**Comment by alexiskhoury** [comment]
Hi David,

Thanks for your prompt reply. Will follow your suggestions.

Regards...
